### PR TITLE
Use cached pioneerid

### PR DESCRIPTION
**Pull request by pdehaan**

### CODE REVIEW

**Review by mythmon — DISMISSED** [review]
This looks good to me.

@rehandalal I think this has the same CI failure as the last PR I filed. Can you restart it again, and maybe see if you can grant me access to do the same, if that's ok?

File: src/PioneerUtils.ts [hunk 1]
```diff
@@ -256,7 +256,7 @@ export class PioneerUtils {
    */
   async chooseBranch() {
     const pioneerId = await this.getPioneerId();
-    const hashKey = `${this.config.studyName}/${this.getPioneerId()}`;
+    const hashKey = `${this.config.studyName}/${pioneerId}`;
```
Comment (pdehaan) [inline src/PioneerUtils.ts]: I have to admit, I'm not 💯 on what I'm doing, but it seemed wasteful to do the lookup twice (throwing away the first lookup). 
The `getPioneerId()` method isn't marked as `async`, so not sure why the line 258 has an `await`. In other places we used the value directly and didn't use `await`. So conversely, maybe just kill line 258 and the temporary `pioneerId` const.

**Comment by mythmon** [comment]
So it turns out it is totally legal to await non Promise values?

```js
async function foo() {
    let x = await bar();
    return x * 2;
}

function bar() {
    return 5;
}

foo().then(v => console.log('ok', v));
// ok 10
```

From MDN:

> If the value of the *expression* following the `await` operator is not a `Promise`, it's converted to a [resolved Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/resolve).

**Comment by mythmon** [comment]
@pdehaan I changed your branch to not include the weird confusing `await` that was there before.

**Review by mythmon — APPROVED** [review]

**Comment by mythmon** [comment]
@rehandalal I could use some help with CI again though.

**Comment by pdehaan** [comment]
I don't seem to have any Circle-CI permissions for this repo, but you could _maybe_ try forcing a "Rebuild without cache" and see if that helps.

continuous_integration_and_deployment

**Pull request closed without merging**